Mod2/Lessons/Week4/CRUDwithORM.md

```diff
@@ -0,0 +1,184 @@
+
+## Create
+
+In the previous lesson, we created the PlantTracker database with two tables: `plants` and `rooms`. There is a one-to-many relationship between rooms and plants: each plant is located in one room, and each room can have many plants. 
+
+Let's add a record to the room table. A room has an ID number (integer) as its primary key, along with a name field (text) and if the room has sunlight (boolean). Our first room is the kitchen that has ample sunlight.
+
+### Room
+
+Let's add a record to the room table. A room has an ID number as its primary key, along with a name field and if the room has sunlight. Our first room is the kitchen that has ample sunlight.
+
+> How would you add this room to your database using SQL statements?
+> `INSERT INTO rooms (id, name, has_sunlight)`
+> `VALUES (1, 'Kitchen', 'true');`
+
+In this example, we have given the room id a value of `1`. Remember that the `id` field automatically increments. This is the first room, so the `id` field of this record will (most likely) be given the value of `1`.
+
+> What happens if there is already a room with an id of `1` in our database?
+
+Let's use Entity Framework to perform this task for us. In `Program.cs`, add the following code in the `Main` method:
+
+  
+
+```c#
+using (var context = new PlantTrackerContext())
+{
+	var kitchen = new Room
+	{
+		Name = "Kitchen",
+		HasSunlight = "true"
+	};
+
+	context.Rooms.Add(kitchen);
+	context.SaveChanges();
+}
+```
+
+  
+
+This code does four things:
+1.  Creates a local `context` object from the `PlantTrackerContext` class. The context object is doing the heavy lifting in terms of connecting the C# objects (Plants and Rooms) to the database.    
+2.  Creates a new Room object in C# called `kitchen`.    
+3.  Adds the `kitchen` to the context object. This step creates an SQL statement to insert the record into the Rooms table.    
+4.  Saves the change to the database.    
+
+In your breakout room, add another room to our application. This time, we need a bedroom that has adequate sunlight. Hint: you can reuse an existing context (i.e. you do not need to make a new context object to add another room).  
+
+### Plant
+
+We now have two rooms. Let's now add a plant. A plant has an ID number as its primary key, a type, a purchase date, and a room. Now it makes sense that we added rooms before adding plants. It also means that we need to select a room for the plant. After much shopping, we purchased a pothos plant on Valentine's Day (February 14, 2023) to add to the kitchen.
+
+> How would you add this plant to your database using SQL statements?
+> `INSERT INTO plants (id, type, purchase_date, room_id)`
+> `VALUES (1, 'Pothos', '2023-02-14 12:00:00', 1);`
+
+Using Entity Framework, add the following code to the using block in the `Main` method in `Program.cs`:
+  
+```c#
+var pothos = new Plant
+{
+	Type = "Pothos"
+	PurchaseDate = "2023-02-14 12:00:00",
```

Comment:
Are these dates working for you? I was running into problems with timezones and had to put a timezone such as `1975-06-15T13:45:30-07:00` in my seed data. But would love to not have to if you found a way around it.